spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -853,7 +854,7 @@ CompletionStage<Void> insertBook(Book book);
 
 
 An _asynchronous repository_ is a repository which declares asynchronous repository methods.
-A Jakarta Data provider is not required to support mixing synchronous and asynchronous repository methods within the same repository interface.
+A mixture of synchronous and asynchronous repository methods can be used within the same repository if all of the asynchronous methods are annotated with the `@Asynchronous` annotation to have the Jakarta Concurrency provider handle the asynchronous behavior. Otherwise, the Jakarta Data provider is not required to support mixing synchronous and asynchronous repository methods within the same repository interface. The `@Asynchronous` annotation must not be used on repositories implemented using reactive streams.
```

Comment:
```suggestion
A repository may declare a mixture of synchronous and asynchronous repository methods if every asynchronous method is annotated with the `@Asynchronous` annotation, so that Jakarta Concurrency provides the asynchronous behavior. 
Otherwise, the Jakarta Data provider is not required to support mixing synchronous and asynchronous repository methods within the same repository interface. 
The `@Asynchronous` annotation must not be used on repositories implemented using reactive streams.
```